### PR TITLE
Add unit tests for newsfeed microservice

### LINKED ISSUE #1471: Add unit tests for newsfeed microservice
### Description

Unit/Functional tests are important to ensure that the implemented methods work as intended

### Acceptance Criteria

- Add the following unit tests to the newsfeed microservice
  - services/newsfeed/shared/twitter.js
    - safeRef
    - getTweetText
    - tweetUrl
    - getImageUrl
  
  - services/export/shared/requestAll.js
    - requestAll

- Add following functional tests to export microservice
  - services/newsfeed/shared/twitter.js
    - tweetMapper

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1471

### How was it solved?
- Add the following unit tests to the newsfeed microservice
    - services/newsfeed/shared/twitter.js
        - [x] safeRef
        - [x] getTweetText
        - [x] tweetUrl
        - [x] getImageUrl
        - [x] tweetMapper
    - services/export/shared/requestAll.js
        - [x] requestAll
- Add following functional tests to export microservice
    - Add the following unit tests to the newsfeed microservice
        - [x] tweetMapper
### How was it tested?
- [x] Local

### CODE REVIEW

File: services/newsfeed/tests/unit/shared/twitter.test.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	});
+
+	it('should throw error when called with undefined object', async () => {
+		expect(() => getTweetText(undefined)).toThrow();
+	});
+});
+
+describe('Test tweetUrl method', () => {
+	it('should return correct url when called with tweet', async () => {
+		jest.spyOn(twitterMethods, 'safeRef').mockReturnValueOnce(tweetObject.entities.urls[0].url);
```
Comment (nagdahimanshu) [inline services/newsfeed/tests/unit/shared/twitter.test.js:171]: I think the spy used here is not really mocking the response. Can you re-verify?

File: services/export/tests/unit/shared/requestAll.test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 2,
+				limit: singleRequestLimit,
+			},
+			{
+				extra_param: 'extra_value',
+				offset: singleRequestLimit * 3,
+				limit: singleRequestLimit,
+			},
+		];
```
Comment (sameersubudhi) [inline services/export/tests/unit/shared/requestAll.test.js:49]: Could be simplified to and defined as a function to 
```suggestion
		const expectedResponseLength = 4;
		const expectedResponse = getResponseOfLength(expectedResponseLength);
```

where `getResponseOfLength` can be as follows:

```
const getResponseOfLength = (n) => new Array(n).fill().map((e, i) => ({
	extra_param: 'extra_value',
	offset: singleRequestLimit * i,
	limit: singleRequestLimit,
}));
```

File: services/export/tests/unit/shared/requestAll.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+	it('should call passed function only once when total limit > single response limit but function return total < single response limit', async () => {
+		const singleRequestLimit = 50;
+		const expectedResponse = [
+			{
+				extra_param: 'extra_value',
+				offset: 0,
+				limit: singleRequestLimit,
+			},
+		];
```
Comment (sameersubudhi) [inline services/export/tests/unit/shared/requestAll.test.js:62]: Adjust accordingly.

File: services/export/tests/unit/shared/requestAll.test.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+				total: singleRequestLimit - 1,
+			},
+		});
+		const response = await requestAll(func2, { limit: singleRequestLimit, extra_param: 'extra_value' }, singleRequestLimit * 10);
+		expect(response).toEqual(expectedResponse);
+	});
+
+	it('should call passed function only once when total limit < single response limit', async () => {
+		const singleRequestLimit = 50;
+		const expectedResponse = [
```
Comment (sameersubudhi) [inline services/export/tests/unit/shared/requestAll.test.js:75]: Adjust accordingly.

File: services/newsfeed/tests/unit/shared/twitter.test.js (hunk 1 above)
Comment (sameersubudhi) [reply services/newsfeed/tests/unit/shared/twitter.test.js:171]: Seems to be working on my system.

**Review by sameersubudhi — APPROVED** [review]
Looks good overall. However, please consider the following.

File: services/newsfeed/tests/unit/shared/twitter.test.js (hunk 1 above)
Comment (nagdahimanshu) [reply services/newsfeed/tests/unit/shared/twitter.test.js:171]: Yes, the tests are passing but the concern here is jest.spyOn() is not really mocking the function instead real implemented is being executed. Tests will pass even with removing L171. So, we can remove all mocks here.

Comment (sameersubudhi) [reply services/newsfeed/tests/unit/shared/twitter.test.js:171]: Yes, I retested it and seems like the spy isn't working. @priojeetpriyom please remove the spy in all the testcases.

File: services/export/tests/unit/shared/requestAll.test.js (hunk 2 above)
Comment (priojeetpriyom) [reply services/export/tests/unit/shared/requestAll.test.js]: Applied.

File: services/export/tests/unit/shared/requestAll.test.js (hunk 4 above)
Comment (priojeetpriyom) [reply services/export/tests/unit/shared/requestAll.test.js]: Applied.

File: services/export/tests/unit/shared/requestAll.test.js (hunk 3 above)
Comment (priojeetpriyom) [reply services/export/tests/unit/shared/requestAll.test.js]: Applied.

File: services/newsfeed/tests/unit/shared/twitter.test.js (hunk 1 above)
Comment (priojeetpriyom) [reply services/newsfeed/tests/unit/shared/twitter.test.js]: Moving the test under functional. Thanks for noticing guys 🙏

Comment (priojeetpriyom) [reply services/newsfeed/tests/unit/shared/twitter.test.js]: Yeah, it is dependant on `safeRef` which we were trying to mock.
https://github.com/LiskHQ/lisk-service/blob/1471-add-unit-tests-for-newsfeed-microservice/services/newsfeed/shared/twitter.js#L48-L52

Comment (sameersubudhi) [reply services/newsfeed/tests/unit/shared/twitter.test.js]: `safeRef` isn't dependant on any external service. Please move the tests back to unit. It should be fine.

Comment (priojeetpriyom) [reply services/newsfeed/tests/unit/shared/twitter.test.js]: Reverted.

**Review by sameersubudhi — APPROVED** [review]

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

**Pull request merged**